packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+
+function buildMethodParamMap(model: RLCModel): Map<string, OperationParameter> {
+  const map = new Map<string, OperationParameter>();
+  (model.parameters ?? []).forEach((p) => {
+    const operatonConcante = getOperationConcate(
+      p.operationName,
+      p.operationGroup,
+      model.options?.sourceFrom
+    );
+    map.set(operatonConcante, p);
+  });
+  return map;
+}
+
+function getOperationConcate(
+  opName: string,
+  opGroup: string,
+  sourceFrom?: string
+) {
+  return sourceFrom === "Swagger"
+    ? opGroup === "" || opGroup === "Client"
+      ? opName
+      : `${opGroup}${opName}`
+    : `${opGroup}_${opName}`;
+}
+
+function buildSchemaObjectMap(model: RLCModel) {
+  // include interfaces
+  const map = new Map<string, Schema>();
+  const allSchemas = (model.schemas ?? []).filter(
+    (o) =>
+      isObjectSchema(o) &&
+      (o as ObjectSchema).usage?.some((u) => [SchemaContext.Input].includes(u))
+  );
+  allSchemas.forEach((o) => {
+    map.set(o.name, o);
+  });
+
+  return map;
+}
+
+function mockParameterTypeValue(
+  type: string,
+  parameterName: string,
+  schemaMap: Map<string, Schema>,
+  path: Set<string> = new Set()
+): string | undefined {
+  type = leaveBracket(type?.trim());
+  let tsType: TypeScriptType | undefined;
+  // Give priority to suggest the ts-type from schema
+  if (schemaMap.has(type)) {
+    tsType = toTypeScriptTypeFromSchema(schemaMap.get(type)!);
+  }
+  // Fall back to suggest ts-type from the type iteself
+  if (!tsType) {
+    tsType = toTypeScriptTypeFromName(type);
+  }
+  switch (tsType) {
+    case TypeScriptType.string:
+      return `'{Your ${leaveStringQuotes(parameterName)}}'`;
+    case TypeScriptType.number:
+      return "123";
+    case TypeScriptType.boolean:
+      return "true";
+    case TypeScriptType.date:
+      return "new Date()";
+    case TypeScriptType.unknown:
+      return `"Unknown Type"`;
+    case TypeScriptType.object: {
+      return mockObjectValues(type, parameterName, schemaMap, path);
+    }
+    case TypeScriptType.array: {
+      return mockArrayValues(type, parameterName, schemaMap, path);
+    }
+    case TypeScriptType.record: {
+      return mockRecordValues(type, parameterName, schemaMap, path);
+    }
+    case TypeScriptType.enum:
+    case TypeScriptType.union: {
+      return mockUnionValues(type, parameterName, schemaMap, path);
+    }
+    case TypeScriptType.constant:
+      return type;
+  }
+  return `undefined /**FIXME */`;
+}
+
+function mockUnionValues(
+  type: string,
+  parameterName: string,
+  schemaMap: Map<string, Schema>,
+  path: Set<string> = new Set()
+) {
+  const schema = schemaMap.get(type);
+  if (schema && schema.enum && schema.enum.length > 0) {
+    const first = schema.enum[0];
+    const itemType = schema.typeName ?? schema.type;
+    if (itemType === "string" || (first.typeName ?? first.type) === undefined) {
+      return `"${first}"`;
+    } else if (itemType === "number") {
+      return `${first}`;
+    }
```

Comment:
boolean? not sure if it's possible.

Reply:
for enum cases I don't think boolean is possible because we only allow string and number as enum values. but for union case i am also not sure, to be honest. the schema info is more like a black hole.  let me have a try first.

Reply:
updated.